File: 프로그래머스/lv2/42747. H－Index/H－Index.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        // citations[i]에서 i값을 증가시키고 논문의 수를 감소시키면서 비교 했을 때 
        // 인용 횟수가 논문의 수 이상이 되었을 때의 논문의 수가 H-Index
        Arrays.sort(citations);
        for(int i =0; i < citations.length;i++)
        {            
            int highCitation = citations.length-i;
            if(citations[i] >= highCitation){                
                answer = highCitation;
                break;               
            }
        }       
        return answer;
    }
}

File: 프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    var answer = citations.length;
    
    citations.sort((a,b) => b - a);
    let len = citations.length;
    
    for(let i = 1; i <= len; i++){
        if(citations[i-1] < i){
            answer = i-1; 
            break;
        }
    }
    return answer;
}